versioned/persist/tests/src/main/java/org/projectnessie/versioned/persist/tests/extension/DatabaseAdapterExtension.java

```diff
@@ -0,0 +1,262 @@
+      throws ParameterResolutionException {
+    Parameter parameter = parameterContext.getParameter();
+
+    DatabaseAdapter databaseAdapter = createAdapterResource(context, parameterContext);
+
+    reinit(databaseAdapter);
+
+    Object assign;
+    if (parameter.getType().isAssignableFrom(DatabaseAdapter.class)) {
+      assign = databaseAdapter;
+    } else {
+      throw new IllegalStateException("Cannot assign to " + parameter);
+    }
+    return assign;
+  }
+
+  static <A extends Annotation> Optional<A> findAnnotation(
+      ExtensionContext context, ParameterContext parameterContext, Class<A> annotation) {
+    Optional<A> opt;
+    if (parameterContext != null) {
+      opt = parameterContext.findAnnotation(annotation);
+      if (opt.isPresent()) {
+        return opt;
+      }
+    }
+    opt = context.getTestMethod().flatMap(m -> Optional.ofNullable(m.getAnnotation(annotation)));
+    if (opt.isPresent()) {
+      return opt;
+    }
+    opt = context.getTestClass().flatMap(m -> Optional.ofNullable(m.getAnnotation(annotation)));
+    return opt;
+  }
+
+  static DatabaseAdapter createAdapterResource(
+      ExtensionContext context, ParameterContext parameterContext) {
+    DatabaseAdapterFactory<?> factory =
+        findAnnotation(context, parameterContext, NessieAdapterName.class)
+            .map(NessieAdapterName::value)
+            .map(DatabaseAdapterFactory::loadFactoryByName)
+            .orElseGet(() -> DatabaseAdapterFactory.loadFactory(x -> true));
+
+    DatabaseAdapterFactory.Builder<?> builder = factory.newBuilder();
+    builder
+        .configure(
+            c ->
+                SystemPropertiesConfigurer.configureFromProperties(
+                    c,
+                    property -> {
+                      List<NessieAdapterConfigItem> configs = new ArrayList<>();
+                      if (parameterContext != null) {
+                        configs.addAll(
+                            parameterContext.findRepeatableAnnotations(
+                                NessieAdapterConfigItem.class));
+                      }
+                      Consumer<AnnotatedElement> collector =
+                          m ->
+                              configs.addAll(
+                                  findRepeatableAnnotations(m, NessieAdapterConfigItem.class));
+                      context.getTestMethod().ifPresent(collector);
+                      context
+                          .getTestClass()
+                          .ifPresent(
+                              cls -> {
+                                for (; cls != Object.class; cls = cls.getSuperclass()) {
+                                  collector.accept(cls);
+                                }
+                              });
+
+                      return configs.stream()
+                          .filter(n -> ("nessie.store." + n.name()).equals(property))
```

Comment:
not sure if it's worth putting this prefix into a static final somewhere so that all the other places that use this also use the same prefix